Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java

```diff
@@ -155,6 +166,13 @@ public void onCreate() {
     public int onStartCommand(Intent intent, int flags, int startId) {
         /* Start the service if the driver is active*/
         startForeground(notificationServiceId, createNotification());
+        String startDistanceCalStr = intent.hasExtra("TRIP_STATUS") ? intent.getStringExtra("TRIP_STATUS") : null;
+        if (!isDistanceCalulation && startDistanceCalStr != null && startDistanceCalStr.equals("started")) {
```

Comment:
Once we start the `fusedLocClientForDistanceCal`. we will not stop in again after ride ended? Because we are not making `isDistanceCalulation` to false. and we are removing removing location updates in onDestroy. so even after the ride ended we are reseting these two finalDistanceWithAcc, finalDistance variables. but we didn't stop the location updates so the distance will be calculated and accumulated with the next ride's distance.